### PR TITLE
ARROW-13447: [Release] Verification script for arm64 and universal2 macOS wheels 

### PR DESCRIPTION
**Pull request by kszucs**
- verification script for arm64 macos wheels
- verification tasks for arm64 macos wheels
- verification tasks for source release on arm64 macos

### CODE REVIEW

**Comment by kszucs** [comment]
@github-actions crossbow submit verify-rc-wheels-macos-* --param release=5.0.0 --param rc=1

**Comment by kszucs** [comment]
@github-actions crossbow submit verify-rc-wheels-macos-* --param release=5.0.0 --param rc=1

**Comment by kszucs** [comment]
@github-actions crossbow submit verify-rc-wheels-macos-arm64 --param release=5.0.0 --param rc=1

**Comment by kszucs** [comment]
@github-actions crossbow submit verify-rc-wheels-macos-arm64 --param release=5.0.0 --param rc=1

**Comment by kszucs** [comment]
@github-actions crossbow submit verify-rc-source-cpp-macos-arm64 --param release=5.0.0 --param rc=1

**Comment by kszucs** [comment]
@github-actions crossbow submit verify-rc-wheels-macos-arm64 --param release=5.0.0 --param rc=1

**Comment by kszucs** [comment]
@github-actions crossbow submit --param release=5.0.0 --param rc=1 verify-rc-wheels-macos-11-*

**Comment by kszucs** [comment]
@github-actions crossbow submit --param release=5.0.0 --param rc=1 verify-rc-source-cpp-macos-arm64

**Comment by kszucs** [comment]
@github-actions crossbow submit wheel-manylinux* wheel-macos*

**Comment by kszucs** [comment]
@github-actions crossbow submit --param release=5.0.0 --param rc=1 verify-rc-source-*-macos-arm64

**Comment by kszucs** [comment]
@github-actions crossbow submit wheel-manylinux*

**Comment by kszucs** [comment]
@github-actions crossbow submit --param release=5.0.0 --param rc=1 verify-rc-source-*-macos-arm64

File: ci/scripts/python_wheel_unix_test.sh [hunk 1]
```diff
@@ -54,16 +67,18 @@ import pyarrow.orc
...
-# Install testing dependencies
-pip install -r ${source_dir}/python/requirements-wheel-test.txt
-
-# Execute unittest
-pytest -r s --pyargs pyarrow
+if [ "${CHECK_UNITTESTS}" == "ON" ]; then
+  # Install testing dependencies
+  pip install -U -r ${source_dir}/python/requirements-wheel-test.txt
+  # Execute unittest, test dependencies must be installed
+  python -c 'import pyarrow; pyarrow.create_library_symlinks()'
```
Comment (kszucs) [inline ci/scripts/python_wheel_unix_test.sh:82]: We should enable the CYTHON tests.

**Comment by kszucs** [comment]
@github-actions crossbow submit --param release=5.0.0 --param rc=1 verify-rc-source-*-macos-arm64

**Comment by kszucs** [comment]
@github-actions crossbow submit wheel-manylinux*

**Comment by kszucs** [comment]
@github-actions crossbow submit --param release=5.0.0 --param rc=1 verify-rc-source-java-macos-arm64

**Comment by kszucs** [comment]
@github-actions crossbow submit wheel-macos-*

**Comment by kszucs** [comment]
@github-actions crossbow submit --param release=5.0.0 --param rc=1 verify-rc-source-*-macos-arm64

**Comment by kszucs** [comment]
@github-actions crossbow submit --param release=5.0.0 --param rc=1 verify-rc-source-ruby-macos-arm64

**Comment by kszucs** [comment]
@github-actions crossbow submit --param release=5.0.0 --param rc=1 verify-rc-wheels-macos-11-*

**Comment by kszucs** [comment]
@github-actions crossbow submit -g wheel

Check that the updated wheel test script works when actually building the wheels.

**Comment by kszucs** [comment]
@github-actions crossbow submit --param release=5.0.0 --param rc=1 verify-rc

Exercise all of the verification builds. All wheel verification builds as well as the newly added source verification builds should pass on Apple Silicon.

**Comment by kszucs** [comment]
@github-actions crossbow submit --param release=5.0.0 --param rc=1  --group verify-rc

Exercise all of the verification builds. All wheel verification builds as well as the newly added source verification builds should pass on Apple Silicon.

**Comment by kszucs** [comment]
@github-actions crossbow submit --param release=5.0.0 --param rc=1 --group verify-rc

Exercise all of the verification builds. All wheel verification builds as well as the newly added source verification builds should pass on Apple Silicon.

**Comment by kszucs** [comment]
@github-actions crossbow submit -g wheel

Check that the updated wheel test script works when actually building the wheels.

**Comment by kszucs** [comment]
Brew has started to fail on installing minio, but that should be unrelated to this PR.

File: dev/release/verify-release-candidate.sh [hunk 2]
```diff
@@ -631,15 +619,42 @@ test_macos_wheels() {
...
     # check the mandatory and optional imports
     pip install --find-links python-rc/${VERSION}-rc${RC_NUMBER} pyarrow==${VERSION}
-    check_python_imports
-
-    # install test requirements and execute the tests
-    pip install -r ${ARROW_DIR}/python/requirements-test.txt
-    python -c 'import pyarrow; pyarrow.create_library_symlinks()'
-    pytest --pyargs pyarrow
+    INSTALL_PYARROW=OFF ARROW_FLIGHT=${check_flight} ARROW_S3=${check_s3} \
+     ${ARROW_DIR}/ci/scripts/python_wheel_unix_test.sh ${ARROW_DIR}
```
Comment (kou) [inline dev/release/verify-release-candidate.sh:623]:
```suggestion
      ${ARROW_DIR}/ci/scripts/python_wheel_unix_test.sh ${ARROW_DIR}
```

File: dev/tasks/verify-rc/github.macos.arm64.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        run: rm -rf arrow
+
+      {{ macros.github_checkout_arrow()|indent }}
+
+      - name: Run verification
+        shell: bash
+        run: |
+          export PATH="/opt/homebrew/opt/node@14/bin:$PATH"
+          export PATH="/opt/homebrew/opt/ruby/bin:$PATH"
+          export PKG_CONFIG_PATH="/opt/homebrew/opt/ruby/lib/pkgconfig"
```
Comment (kou) [inline dev/tasks/verify-rc/github.macos.arm64.yml:45]:
```suggestion
          export PATH="$(brew --prefix node@14)/bin:$PATH"
          export PATH="$(brew --prefix ruby)/bin:$PATH"
          export PKG_CONFIG_PATH="$(brew --prefix ruby)/lib/pkgconfig"
```

**Review by kou — APPROVED** [review]
+1

**Pull request closed without merging**